pkg/daemon/update.go

```diff
@@ -711,12 +711,20 @@ func (dn *Daemon) updateHypershift(oldConfig, newConfig *mcfgv1.MachineConfig, d
 // We don't generally expect administrators to use this versus e.g. removing
 // broken configuration. We only remove the rollback once the MCD pod has
 // landed on a node, so we know kubelet is working.
-func (dn *Daemon) removeRollback() error {
+func (dn *Daemon) removeRollback() {
```

Comment:
So in theory, without this being successful, we should be retrying from the main sync loop anyways, although I guess in practice, that might not be the case?

I guess I am a bit concerned with the daemon potentially just endlessly blocking here without explicitly erroring anything up the stack. Could we maybe have a cap on the # of retries? (e.g. 10 times)

Reply:
You're absolutely right, we *should* be retrying this.   In the linked logs what I see is:

```
2022-10-26T10:14:27.312249160Z I1026 10:14:27.312203    2425 update.go:2091] Running: rpm-ostree cleanup -r
2022-10-26T10:14:52.619031013Z E1026 10:14:52.618989    2425 writer.go:200] Marking Degraded due to: Failed to remove rollback: error running rpm-ostree cleanup -r: error: Timeout was reached
2022-10-26T10:14:52.619031013Z : exit status 1
2022-10-26T10:14:52.658269675Z I1026 10:14:52.658216    2425 daemon.go:511] Transitioned from state: Done -> Degraded
2022-10-26T10:14:52.717753505Z I1026 10:14:52.717711    2425 daemon.go:1220] In bootstrap mode
2022-10-26T10:14:52.717753505Z I1026 10:14:52.717734    2425 daemon.go:1248] Current+desired config: rendered-worker-e215579a626ff9cf73fe8560b6f9aca4
2022-10-26T10:14:52.717753505Z I1026 10:14:52.717739    2425 daemon.go:1258] state: Degraded
2022-10-26T10:14:52.773492704Z I1026 10:14:52.773448    2425 update.go:2091] Running: rpm-ostree cleanup -r
```

Which...could be us getting stuck for some other reason, or it could be hmm, now what we're hitting is the lack of having the node journals in must-gather :cry: 

I'm not yet able to get access to any other logs.

I wonder here...is what we're really doing here papering over transiently going degraded during an upgrade?  But the next question then is "how transient is this"?